File: src/main/java/pl/funnyskaddon/core/fix/PointsChangeListener.java
package pl.funnyskaddon.core.fix;

import net.dzikoysk.funnyguilds.event.FunnyEvent.EventCause;
import net.dzikoysk.funnyguilds.event.rank.PointsChangeEvent;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;

public class PointsChangeListener implements Listener{
	@EventHandler
	public static void onRank(final PointsChangeEvent e) {
		if(e.getEventCause() == EventCause.USER) {
			int change = e.getChange();
			Player attacker = e.getDoer().getPlayer();
			Player victim = null;
			if(!e.getDoer().equals(e.getRank().getUser())) {
				victim = e.getRank().getUser().getPlayer();
			}
			if(victim != null && attacker != null) {
				final KillPointsChangeEventFix event = new KillPointsChangeEventFix(attacker, victim, change);
				Bukkit.getServer().getPluginManager().callEvent(event);
			}
		}
	}

}

File: src/main/kotlin/pl/funnyskaddon/events/rank/PointsChangeListener.kt
package pl.funnyskaddon.events.rank

import net.dzikoysk.funnyguilds.event.FunnyEvent
import net.dzikoysk.funnyguilds.event.rank.PointsChangeEvent
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import pl.funnyskaddon.FunnySkAddon

class PointsChangeListener(private val plugin: FunnySkAddon) : Listener {

    @EventHandler
    fun onChange(event: PointsChangeEvent) {
        if (event.eventCause != FunnyEvent.EventCause.USER) {
            return
        }

        val change = event.change
        val attacker = event.doer.player

        if (event.doer == event.rank.user) {
            return
        }
        val victim = event.rank.user.player

        if (attacker == null || victim == null) {
            return
        }
        plugin.server.pluginManager.callEvent(CustomKillPointsChangeEvent(event.eventCause, attacker, victim, change, event.rank))
    }

}